packages/apollo-link-http/src/__tests__/sharedHttpTests.ts

```diff
@@ -842,9 +842,30 @@ export const sharedHttpTest = (
 
     const body = '{';
     const unparsableJson = jest.fn(() => Promise.resolve(body));
-    it('throws an error if response is unparsable', done => {
+    it('throws an error if response is unparsable and status >= 300', done => {
```

Comment:
>= 500? your PR sais 5xx responses

Reply:
Well in the codebase I noticed there is only distinction between status codes 2xx and others (not successful). I will change the PR description.